src/chat.rs

```diff
@@ -4106,20 +4104,17 @@ pub async fn forward_msgs(context: &Context, msg_ids: &[MsgId], chat_id: ChatId)
         bail!("cannot send to {}: {}", chat_id, reason);
     }
     curr_timestamp = create_smeared_timestamps(context, msg_ids.len());
-    let ids = context
-        .sql
-        .query_map(
-            &format!(
-                "SELECT id FROM msgs WHERE id IN({}) ORDER BY timestamp,id",
-                sql::repeat_vars(msg_ids.len())
-            ),
-            rusqlite::params_from_iter(msg_ids),
-            |row| row.get::<_, MsgId>(0),
-            |ids| ids.collect::<Result<Vec<_>, _>>().map_err(Into::into),
-        )
-        .await?;
-
-    for id in ids {
+    let mut msgs = Vec::with_capacity(msg_ids.len());
+    for id in msg_ids {
+        let ts: i64 = context
+            .sql
+            .query_get_value("SELECT timestamp FROM msgs WHERE id=?", (id,))
```

Comment:
Can be optimized using `Sql::call()` + preparing a statement and executing it in a loop. But below there's a heavier loop calling `Message::load_from_db()` anyway